File: C++/inicio/InicioYFin.cpp
#include <iostream>
#include <vector>
#include <string>
using namespace std;

void imprimir(const vector<string>& v)
{
    for (size_t i = 0; i < v.size(); i++)
    {
        cout << v[i];
        if (i != v.size() - 1) cout << ", ";
    }
    cout << endl;
}

int main()
{
    // Insercion al inicio
    vector<string> abecedario = {"a", "b", "c", "d"};
    imprimir(abecedario);
    // muestra normal

    abecedario.insert(abecedario.begin(), "z");
    imprimir(abecedario);

    // Insercion al final
    abecedario = {"a", "b", "c", "d"};
    imprimir(abecedario);
    // muestra normal

    abecedario.insert(abecedario.begin() + 4, "z");
    imprimir(abecedario);

    return 0;
}

File: Java/inicio/InicioYFin.java
import java.util.ArrayList;
import java.util.List;

public class InicioYFin {
    public static void main(String[] args) {
        // Insercion al inicio
        List<String> abecedario = new ArrayList<>(List.of("a", "b", "c", "d"));
        System.out.println(abecedario);
        // muestra normal

        abecedario.add(0, "z");
        System.out.println(abecedario);

        // Insercion al final
        abecedario = new ArrayList<>(List.of("a", "b", "c", "d"));
        System.out.println(abecedario);
        // muestra normal

        abecedario.add(4, "z");
        System.out.println(abecedario);
    }
}
